src/webgpu/shader/execution/shader_io/fragment_builtins.spec.ts

```diff
@@ -779,3 +793,83 @@ g.test('inputs,interStage')
       })
     );
   });
+
+g.test('inputs,sample_index')
+  .desc(
+    `
+    Test fragment shader builtin(sample_index) values.
+  `
+  )
+  .params(u =>
+    u //
+      .combine('nearFar', [[0, 1] as const, [0.25, 0.75] as const] as const)
+      .combine('sampleCount', [1, 4] as const)
+      .combine('interpolation', [
+        { type: 'perspective', sampling: 'center' },
+        { type: 'perspective', sampling: 'centroid' },
+        { type: 'perspective', sampling: 'sample' },
+        { type: 'linear', sampling: 'center' },
+        { type: 'linear', sampling: 'centroid' },
+        { type: 'linear', sampling: 'sample' },
+        { type: 'flat' },
+      ] as const)
+  )
+  .beforeAllSubcases(t => {
+    const {
+      interpolation: { type, sampling },
+    } = t.params;
+    t.skipIfInterpolationTypeOrSamplingNotSupported({ type, sampling });
+  })
+  .fn(async t => {
+    const {
+      nearFar,
+      sampleCount,
+      interpolation: { type, sampling },
+    } = t.params;
+    // prettier-ignore
+    const clipSpacePoints = [       // ndc values
+      [0.333, 0.333, 0.333, 0.333],  //  1,  1, 1
+      [ 1.0,  -3.0,   0.25, 1.0  ],  //  1, -3, 0.25
+      [-1.5,   0.5,   0.25, 0.5  ],  // -3,  1, 0.5
+    ];
+
+    const interStagePoints = [
+      [1, 2, 3, 4],
+      [5, 6, 7, 8],
+      [9, 10, 11, 12],
+    ];
+
+    const width = 4;
+    const height = 4;
+    const actual = await renderFragmentShaderInputsTo4TexturesAndReadbackValues(t, {
```

Comment:
Question for the `renderFragmentShaderInputsTo4TexturesAndReadbackValues` function implementation: https://github.com/gpuweb/cts/blame/ae15a59832989c22982acaeaccdf5d379afced62/src/webgpu/shader/execution/shader_io/fragment_builtins.spec.ts#L489

`let u = bitcast<vec4u>(v);`

Is it bitcast or typecast? I think if turning f32 value 1.0 to u32 value 1u, it's `let u = vec4u(v)`?

Reply:
It's a bitcast. The code is writing four f32 values into for rgba8unorm textures, one f32 to each rgba8unorm pixel in each of the 4 textures. See the comment here: https://github.com/gpuweb/cts/blob/67c1786ad65fb9ba2e8664e442d1a25d7c823ebe/src/webgpu/shader/execution/shader_io/fragment_builtins.spec.ts#L399